src/main/webapp/jsp/studentCourseJoinConfirmation.jsp

```diff
@@ -15,18 +15,47 @@
 	    </div>
 	    <div class="panel-body">
 	        <p>
-	            You are currently logged in as <span><strong>${data.account.googleId}</strong></span>. 
-	            <br>If this is not you please <a href="${data.logoutUrl}">log out</a> and re-login using your own Google account.
-	            <br>If this is you, please confirm below to complete your registration.
+	            You are currently logged in as <span><strong>${data.account.googleId}</strong></span>.
+	            <c:if test="${data.redirectResult}">
+	                You have been redirected to this page because you navigated to a link for the course <strong>${data.courseId}</strong>, which you have not been registered in.
+	            </c:if>
 	            <br>
+	            <c:choose>
+	                <c:when test="${data.redirectResult}">
+	                    If you wish to register as <strong>${data.account.googleId}</strong>, please confirm below to complete your registration.
+	                    <br>If you wish to register with another Google ID or do not wish to register for this course, please
+	                    <a href="${data.logoutUrl}">log out</a>
+	                    , log in with your desired Google ID if necessary, and navigate to the link sent to you via email again.
```

Comment:
Now comes the distinction between can access without registering and can access without login. An easy way out is to make the words "or do not wish to register for this course" and "if necessary" spelled out only if the next URL is accessible without login.
And I'm thinking to remove "sent to you via email" because for all we know the link may come from somewhere else.
Sounds good?